File: generated-code/example/java/src/main/java/model/Enumeration.java
package model;

import util.StreamUtil;

public enum Enumeration {
    VALUE_ONE(0),
    VALUE_TWO(1);

    public int tag;

    Enumeration(int tag) {
        this.tag = tag;
    }

    public static Enumeration readFrom(java.io.InputStream stream) throws java.io.IOException {
        switch (StreamUtil.readInt(stream)) {
        case 0:
            return VALUE_ONE;
        case 1:
            return VALUE_TWO;
        default:
            throw new java.io.IOException("Unexpected tag value");
        }
    }
}

File: generated-code/example/kotlin/src/main/kotlin/model/Enumeration.kt
package model

import util.StreamUtil

enum class Enumeration private constructor(val tag: Int) {
    VALUE_ONE(0),
    VALUE_TWO(1);

    companion object {
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Enumeration {
            return when (StreamUtil.readInt(stream)) {
            VALUE_ONE.tag -> VALUE_ONE
            VALUE_TWO.tag -> VALUE_TWO
            else -> throw java.io.IOException("Unexpected tag value")
            }
        }
    }
}